enabler/src/components/payment-methods/invoice/invoice.ts

```diff
@@ -0,0 +1,54 @@
+import { ComponentOptions, PaymentMethod } from '../../../payment-enabler/payment-enabler';
+import styles from '../../../style/style.module.scss';
+import { BaseComponent, BaseOptions } from '../../base';
+
+export class Invoice extends BaseComponent {
+  constructor(baseOptions: BaseOptions, componentOptions: ComponentOptions) {
+    super(PaymentMethod.invoice, baseOptions, componentOptions);
+  }
+
+  mount(selector: string) {
+    document.querySelector(selector).insertAdjacentHTML("afterbegin", this._getTemplate());
+    document.querySelector('#invoiceForm').addEventListener('onsubmit', (e) => {
+      e.preventDefault();
+      this.submit();
+    });
+
+  }
+
+  async submit() {
+    // here we would call the SDK to submit the payment
+    this.sdk.init({ environment: this.environment });
+    try {
+      const requestData = {
+        paymentMethod: {
+            type: this.paymentMethod,
+        }
+      };
+      const response = await fetch(this.processorUrl + '/payments', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json', 'X-Session-Id': this.sessionId },
+        body: JSON.stringify(requestData),
+      });
+      const data = await response.json();
+
+      if (data.outcome === 'Authorized' && data.paymentReference) {
+        this.onComplete && this.onComplete({ isSuccess: true, paymentReference: data.paymentReference });
+      } else {
+        this.onComplete && this.onComplete({ isSuccess: false });
+      }
+    } catch(e) {
+      this.onError('Some error occurred. Please try again.');
+    }
+  }
+
+  private _getTemplate() {
+  return `
+    <div class="${styles.wrapper}">
+      <form id="invoiceForm" class="${styles.paymentForm}">
+      Pay easily with Invoice and transfer the shopping amount within the specified date.
```

Comment:
Do we need to add button for form submission here?

Reply:
We don't have submit button for the Invoice connector. 
We have added submit on FormElement.
 I will document it, So that users knows this works on Form submission rather than button.